runner/src/server/services/uploadService.ts

```diff
@@ -159,6 +162,24 @@ export class UploadService {
       files = this.fileStreamsFromPayload(request.payload);
     }
 
+    const clientSideUploadComponent = page.components.items.find(
+      (c) => c.type === "ClientSideFileUploadField"
+    );
+    if (clientSideUploadComponent && applicationId) {
+      const { id, path } = request.params;
+      const delPath = `${applicationId}/${id}/${path}/${clientSideUploadComponent.name}`;
+      const filesToDelete =
+        request.payload[`${clientSideUploadComponent.name}__delete[]`] || [];
+
+      if (Array.isArray(filesToDelete)) {
+        for (const fileKeyToDelete of filesToDelete) {
+          await this.deleteFileS3(`${delPath}/${fileKeyToDelete}`);
+        }
+      } else {
+        await this.deleteFileS3(`${delPath}/${filesToDelete}`);
+      }
```

Comment:
Hapi will intuitively interpret this as an array if it's multiple files, or a single file if not - we need a condition to handle this.